### DISCUSSION TITLE
Workflow for using local firmware

### DISCUSSION THREAD

**Discussion by jonesea** [dl9rdz/rdz_ttgo_sonde#391] [TTGO firmware]
What's everyone's workflow for using their own complied firmware?

I'm using the Arduino IDE method as that's what I'm more familiar with.  I compiled from untouched firmware successfully, but uploading via the traditional Arduino serial yielded a soft-bricked LoRa32 board (it was first loaded with the full dev branch image from git).

Ideally, I would like to have upgrade images so that I can keep my settings on the SPIFF partition.

I also didn't see a traditional way of working with local upgrade images via the built-in web based update, I even tried modifying update.html file with local references and a local web server to host the image... But there seems to be some hardcoded behaviour in the source that prevent this for working.

What are some ways to upload locally generated firmware?

Apologies if this is obvious...

**Comment by hr-ru** [comment]
Currently this *should* be no problem at all.

If you just "run" (upload via serial) in Arduino IDE (or, using PlatformIO, do a "pio run --target=upload" on the command line, or the equivalent in the GUI), then this is equivalent to just an upgrade, leaving the SPIFFS partition unmodified. 

(For also flashing the SPIFSS, you have to manually do this either in Arduino IDE or pio run target=uploadfs)

**Reply by hr-ru** [reply]
If you want to do over-the-air-updates from your own images, you need to modify the source, the update host is hard coded:
https://github.com/dl9rdz/rdz_ttgo_sonde/blob/e5c2e2db7745fcb40f8bbe1a1c596ecc88c83d1a/RX_FSK/RX_FSK.ino#L59

**Comment by jonesea** [comment]
Tried again, and I succeed.  I'm unsure why I had failures before.

Thank you for the hint on the local web upload.
